account/src/main/java/com/user/account/controller/AllControlers.java

```diff
@@ -28,15 +37,19 @@ public String home() {
 		// Updating accountBalance after debit
 		
 		
-		@PostMapping("/debit/{accountNumber}/{amount}")
-		public User debit(@PathVariable String accountNumber, @PathVariable String amount) {
+		@PostMapping("/debit/{accountNumber}")
+		public User debit(@PathVariable String accountNumber, @RequestBody String amount) {
 			
 			User user = this.accountServices.getUser(Long.parseLong(accountNumber));
 			
 			if ( Long.parseLong(amount) <= user.getAccountBalance() ) {
 				
 				user.updateBalanceBy(-Long.parseLong(amount)) ;
 				
+				Transactions newTransaction = new Transactions(user.getAccountNumber(), Long.parseLong(amount), "debit");
```

Comment:
Avoid `new`, try to use factory for creating objects as per DI principal.

Reply:
Using new will avoid the problem of concurrency. Since we need to have a different transaction number for different transactions.

Reply:
?